Add FAQs and additional documentation (#138)
connectrpc/connectrpc.com: docs/faq.md

We encourage you to give it a try and report any issues you find to Go or
connect-go to help bring WASM in Go forward.

### Why am I seeing a "stream error: stream ID 5; INTERNAL_ERROR; received from peer" error message after X seconds?

The Go HTTP server has a read or write timeout configured. This timeout applies
to the entire operation duration, even for streaming calls. So if an operation
takes longer than that, the server closes the stream and clients can see the
above error message.

### How do I close a client response stream in connect-go?

On reading the response, a client calling `CloseResponse` will attempt to
gracefully close the connection. `Close` does this by reading all of the
remaining messages sent from the server until the final status message is
received. If the status is in error, `Close` will return the wire error.
Alternatively, if you wish to cancel the operation and immediately stop
the client stream, see [below](#cancel-stream) to cancel the connection.

### How do I cancel a client response stream in Connect-go?

To cancel and abort a stream, call the cancel function of the underlying

> context associated with the connection. This context is provided on stream

Reviewer: ```suggestion
context associated with the stream. This context is provided on stream
```

> creation. On cancel, the stream is aborted and any resources along with it.

Reviewer: ```suggestion
creation. On cancel, the stream is aborted and any resources are released along with it.
```

We encourage you to give it a try and report any issues you find to Go or
connect-go to help bring WASM in Go forward.

### Why am I seeing a "stream error: stream ID 5; INTERNAL_ERROR; received from peer" error message after X seconds? {#stream-error}

It means that your [http.Server](https://pkg.go.dev/net/http#Server) has a
`ReadTimeout` or `WriteTimeout` field configured. These fields apply
to the entire operation duration, even for streaming calls. If an operation
takes longer than the value specified, the server closes the stream and
clients can see the above error message. The other timeout fields won't cause
this error, and we

> ncourage you to set `ReadHeaderTimeout` in particular.

Reviewer: ```suggestion
encourage you to set `ReadHeaderTimeout` in particular.
```

### How do I close a client response stream in connect-go?

> On reading the response, a client calling `CloseResponse` will attempt to
> gracefully close the connection. `Close` does this by reading all of the
> remaining messages sent from the server until the final status message is
> received. If the status is in error, `Close` will return the wire error.

Reviewer: We have a mix of calling it `CloseResponse` and `Close`. It's called `CloseResponse` for bidi streams and `Close` for server streams. Maybe we can clarify that and then just refer to it like "close" (without the fixed-width code font).
```suggestion
On reading the response, a client can call `CloseResponse` on bidirectional
streams or `Close` on server streams to gracefully close the connection.
This will discard any remaining messages sent from the server until the
final status message is received. If the status is an error, the close function
will return the wire error.
```

Alternatively, if you wish to cancel the operation and immediately stop

> the client stream, see [below](#cancel-stream) to cancel the connection.

Reviewer: ```suggestion
the client stream, see [below](#cancel-stream) to cancel the operation.
```
